internal/clog/error.go

```diff
@@ -0,0 +1,133 @@
+package clog
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/fatih/color"
+	"golang.org/x/xerrors"
+)
+
+// CLIMessage provides a human-readable message for CLI errors and messages.
+type CLIMessage struct {
+	Level  string
+	Color  color.Attribute
+	Header string
+	Lines  []string
+}
+
+// CLIError wraps a RichCLIMessage a allows consumers to treat it as a normal error.
+type CLIError struct {
+	CLIMessage
+	error
+}
+
+// Error formats the CLI message for consumption by a human as an error.
+func (r CLIMessage) String() string {
+	var str strings.Builder
+	str.WriteString(fmt.Sprintf("%s: %s\n",
+		color.New(r.Color).Sprint(r.Level),
+		color.New(color.Bold).Sprint(r.Header)),
+	)
+	for _, line := range r.Lines {
+		str.WriteString(fmt.Sprintf("  %s %s\n", color.New(r.Color).Sprint("|"), line))
+	}
+	return str.String()
+}
+
+// Log logs the given error to stderr, defaulting to "fatal" if the error is not a RichCLIError.
+// If the error is a RichCLIError, the plain error chain is ignored and the rich error
+// is logged on its own.
+func Log(err error) {
+	var cliErr CLIError
+	if !xerrors.As(err, &cliErr) {
+		cliErr = Fatal(err.Error())
+	}
+	fmt.Fprintln(os.Stderr, cliErr.String())
+}
+
+// LogInfo prints the given info message to stderr.
+func LogInfo(header string, lines ...string) {
```

Comment:
I'm a bit confused why `LogInfo` and `LogSuccess` are the only ones prefixed with `Log`

Reply:
Yeah this is definitely confusing. The issue I'm having is that `Error`, `Warn`, and `Fatal` return errors. So it would potentially even more confusing for these to be the only ones with side-effects. 

It gets pretty tricky when we actually _do_ want to log errors as they happen: for example when iterating through a series of `stop` actions.

Any ideas?

Reply:
Ohh i see now, I was skimming over the returns. Considering the others return errors this makes more sense